src/os/bluestore/BlueStore.cc

```diff
@@ -7445,6 +7454,11 @@ void BlueStore::_txc_update_store_statfs(TransContext *txc)
   logger->inc(l_bluestore_compressed_allocated, txc->statfs_delta.compressed_allocated());
   logger->inc(l_bluestore_compressed_original, txc->statfs_delta.compressed_original());
 
+  {
+    std::lock_guard<std::mutex> l(vstatfs_lock);
```

Comment:
@varadakari - No, I don't expect any performance difference here - single lock acquisition is a pretty simple operation, we have tons of them along the write path. This was mainly to resolve Sage's complaint on significant amount of time spent in BlueStore::statfs func